src/Controls/src/Core/AnimationExtensions.cs

```diff
@@ -260,8 +273,24 @@ static void AnimateInternal<T>(IAnimatable self, IAnimationManager animationMana
 
 			s_animations[key] = info;
 
+			if (self is VisualElement ve)
```

Comment:
I think we can unhook from everything inside HandleTweenerFinished

<img width="897" alt="image" src="https://github.com/user-attachments/assets/f2a986d9-4d25-4b94-a0f6-a2bb952699d7">